### ISSUE TITLE
inconsistent initial System for Modelica.Electrical.Machines.Examples.SynchronousInductionMachines.SMEE_LoadDump

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1570]
**Reported by jfrenkel on 19 Sep 2014 09:44 UTC**
The MSL example Modelica.Electrical.Machines.Examples.SynchronousInductionMachines.SMEE_LoadDump has an inconsisten initial System. 

The Variables `switch.closerWithArc[1].tSwitch`,`switch.closerWithArc[2].tSwitch` and `switch.closerWithArc[3].tSwitch` cannot calculated by the initial system because in the assigned equation they are not in the activ branch:
```
    // Unknowns[1]: 
    //   switch.closerWithArc[1].tSwitch(start = -1E+060)

  equation // Residual equations
    0 = (if loadControl.y then switch.closerWithArc[1].v-switch.closerWithArc[1].Ron
      *loadInductor.inductor[1].i else (if switch.closerWithArc[1].quenched
       then loadInductor.inductor[1].i-switch.closerWithArc[1].Goff*
      switch.closerWithArc[1].v else switch.closerWithArc[1].v-min(
      switch.closerWithArc[1].Vmax, switch.closerWithArc[1].V0+switch.closerWithArc[1].dVdt
      *(time-switch.closerWithArc[1].tSwitch))*noEvent((if loadInductor.inductor[1].i
       > 0 then 1 else (if loadInductor.inductor[1].i < 0 then -1 else 0)))));
```
`loadControl.y=false` and `switch.closerWithArc[1].quenched==true`.
the same for the other variables. 

For example Dymola reports in the log file: 
```
ERROR: Failed to solve non-linear system using Newton solver.
To get more information: Turn on Simulation/Setup/Debug/Nonlinear solver diagnostics/Details
Solution to systems of equations not found at time = 0
   Nonlinear system of equations number = 1
   Infinity-norm of residue = 8.73556E-008
   The tolerance of 5E-011
   is too small. No further improvement
   in the approximate solution is possible.
   Accumulated number of residue       calc.: 44
   Accumulated number of symbolic Jacobian calc.: 21
   Last values of solution vector:
switch.closerWithArc[2].tSwitch = -1E+060
   Last values of residual vector:
{ -8.73556E-008 }
 
Trying to solve non-linear system using global homotopy-method.
ERROR: Failed to solve non-linear system using Newton solver.
To get more information: Turn on Simulation/Setup/Debug/Nonlinear solver diagnostics/Details
Solution to systems of equations not found at time = 0
   Nonlinear system of equations number = 2
   Infinity-norm of residue = 1.32795E-007
   The tolerance of 5E-011
   is too small. No further improvement
   in the approximate solution is possible.
   Accumulated number of residue       calc.: 44
   Accumulated number of symbolic Jacobian calc.: 21
   Last values of solution vector:
switch.closerWithArc[1].tSwitch = -1E+060
   Last values of residual vector:
{ -1.32795E-007 }

Scaled residual 1.327954502406036E-007
Old Residual 1.327954502406036E-007
Predicted relative decrease 0
Actual relative decrease 0
Cannot reach precision since step bound is 5.980575402200798E-053
  Solution: Infinity-norm of residue = 1.32795E-007
            Acc. number of residue     calc.: 2764
            Acc. number of symbolic Jacobian calc.: 232
switch.closerWithArc[1].tSwitch = -1E+060
Residual:
{ -1.32795E-007 }
```
I have no idea why Dymola accept this.

But also OpenModelica accept it: 
```
SingleEquation: 128
1: switch.closerWithArc[2].tSwitch:DISCRETE(start = -9.999999999999999e+059 fixed = false protected = true ) .Modelica.Electrical.Machines.Examples.SynchronousInductionMachines.SMEE_LoadDump, .Modelica.Electrical.MultiPhase.Ideal.CloserWithArc, .Modelica.Electrical.Analog.Ideal.CloserWithArc, .Modelica.SIunits.Time type: Real [3]
1/1 (1): 0.0 = if not switch.closerWithArc[2].off then switch.closerWithArc[2].v - switch.closerWithArc[2].Ron * loadInductor.inductor[2].i else 
        -(if switch.closerWithArc[2].quenched then loadInductor.inductor[2].i - switch.closerWithArc[2].Goff * switch.closerWithArc[2].v else switch.closerWithArc[2].v - min(switch.closerWithArc[2].Vmax, switch.closerWithArc[2].V0 + switch.closerWithArc[2].dVdt * (time - switch.closerWithArc[2].tSwitch)) * /*Real*/(sign(loadInductor.inductor[2].i)))   [unknown]

	|                 | |       | | | [47] Real loadInductor.inductor[2].i(start=0, nominal=1) = -0 (pre: 0)
|                 | |       | | | [265] parameter Real switch.closerWithArc[2].Goff(start=0, fixed=true) = 1e-005	
|                 | |       | | | [191] Real switch.closerWithArc[2].v(start=0, nominal=1) = -0.00454398 (pre: 0)
Residual:
0.0 = - (0.0 - 1e-005*-0.00454398)
0.0 = -0,0000000454398


SingleEquation: 123
1: switch.closerWithArc[3].tSwitch:DISCRETE(start = -9.999999999999999e+059 fixed = false protected = true ) .Modelica.Electrical.Machines.Examples.SynchronousInductionMachines.SMEE_LoadDump, .Modelica.Electrical.MultiPhase.Ideal.CloserWithArc, .Modelica.Electrical.Analog.Ideal.CloserWithArc, .Modelica.SIunits.Time type: Real [3]
1/1 (1): 0.0 = if not switch.closerWithArc[2].off then switch.closerWithArc[3].v - switch.closerWithArc[3].Ron * loadInductor.inductor[3].i else 
        -(if switch.closerWithArc[3].quenched then loadInductor.inductor[3].i - switch.closerWithArc[3].Goff * switch.closerWithArc[3].v else switch.closerWithArc[3].v - min(switch.closerWithArc[3].Vmax, switch.closerWithArc[3].V0 + switch.closerWithArc[3].dVdt * (time - switch.closerWithArc[3].tSwitch)) * /*Real*/(sign(loadInductor.inductor[3].i)))   [unknown]

		
|                 | |       | | | [48] Real loadInductor.inductor[3].i(start=0, nominal=1) = 0 (pre: 0)	
|                 | |       | | | [266] parameter Real switch.closerWithArc[3].Goff(start=0, fixed=true) = 1e-005
|                 | |       | | | [192] Real switch.closerWithArc[3].v(start=0, nominal=1) = -0.0132795 (pre: 0)
Residual:
0.0 = - (0.0 - 1e-005*-0.0132795)
0.0 = -0,000000132795
```

My proposal is to change a view fixed selections:
from:
```
Machines.Sensors.VoltageQuasiRMSSensor voltageQuasiRMSSensor(
      ToSpacePhasor1(y(each start=1E-3, each fixed=true)));
```
to:
```
Machines.Sensors.VoltageQuasiRMSSensor voltageQuasiRMSSensor;
```
and from:
```
  Modelica.Electrical.MultiPhase.Ideal.CloserWithArc switch(     m=m,
    Ron=fill(1e-5, m),
    Goff=fill(1e-5, m),
    V0=fill(30, m),
    dVdt=fill(10e3, m),
    Vmax=fill(60, m));
```
to:
```
  Modelica.Electrical.MultiPhase.Ideal.CloserWithArc switch(     m=m,
    Ron=fill(1e-5, m),
    Goff=fill(1e-5, m),
    V0=fill(30, m),
    dVdt=fill(10e3, m),
    Vmax=fill(60, m),
    closerWithArc(tSwitch(each fixed=true)));
```

With this changes the simulation results are still the same.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1570

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 7 Oct 2014 15:40 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by leo&#46;gall on 6 Jan 2015 19:37 UTC**
Side note:
Translation of this model fails in Dymola 2014 FD01 and Dymola 2015 FD01 when using [maintenance branch](https://svn.modelica.org/projects/Modelica/branches/maintenance/3.2.1) 6a6c175be696b7e83963bc97977bd8c5079b196e.

Translation log message:
```
The initialization problem is overspecified for variables of element type Real.
There are 2 too many scalar conditions.
To correct it you can
inactivate start values for
smee.ir[1](start = 0)
smee.ir[2](start = 0)
speed.phi(start = 0)
switch.closerWithArc[1].tSwitch(start = -1E+060)
switch.closerWithArc[2].tSwitch(start = -1E+060)
switch.closerWithArc[3].tSwitch(start = -1E+060)
voltageQuasiRMSSensor.ToSpacePhasor1.y[1](start = 0.001)
voltageQuasiRMSSensor.ToSpacePhasor1.y[2](start = 0.001)

remove initial equations:
voltageController.I.y = voltageController.I.y_start;
smee.lesigma.i = 0;
smee.idq_sr[2] = 0.0;
smee.idq_sr[1] = 0.0;
speed.w = speed.w_ref;
```

**Comment by modelica-trac-importer** [comment]
**Comment by anonymous on 7 Jan 2015 19:10 UTC**
I do not understand the original report since in Electrical.Analog.Ideal.Oner/CloseWithArc the initial conditions are consistent defined with fixed=true:
```
  Boolean off(start=false, fixed=true);
  discrete Modelica.SIunits.Time tSwitch(start=-Modelica.Constants.inf, fixed=true);
  Boolean quenched(start=true, fixed=true);

```
Maybe this has been fixed since the report?

I understand Leo's remark, I'm working to fix this.

**Comment by modelica-trac-importer** [comment]
**Comment by ahaumer on 7 Jan 2015 19:40 UTC**
Sorry for the anonymous reply, I didn't login. 
I fixed now Leo's issue with f7601d72cf5bbc9f6960058344e6a38ffe3e8056, c80e6a8e7f13cf1d062c3d98c420aa20cbd91f53. 
I'm pretty sure the original issue was fixed before which in turn triggered the new issue (reported by Leo).
